### DISCUSSION TITLE
Get the error : ip_not_available

### DISCUSSION THREAD

**Discussion by K-MeLeOn** [kube-hetzner/terraform-hcloud-kube-hetzner#813] [Q&A]
Hi,

Anyone have any idea why 2/3 nodes are not attached to the network?
When I try to attach them manually, I get the same error message.
A similar issue: #59 

Thanks in advance for your time !

Error:
> module.kube-hetzner.null_resource.configure_autoscaler[0] (remote-exec): serviceaccount/cluster-autoscaler created
> module.kube-hetzner.null_resource.configure_autoscaler[0] (remote-exec): clusterrole.rbac.authorization.k8s.io/cluster-autoscaler created
> module.kube-hetzner.null_resource.configure_autoscaler[0] (remote-exec): role.rbac.authorization.k8s.io/cluster-autoscaler created
> module.kube-hetzner.null_resource.configure_autoscaler[0] (remote-exec): clusterrolebinding.rbac.authorization.k8s.io/cluster-autoscaler created
> module.kube-hetzner.null_resource.configure_autoscaler[0] (remote-exec): rolebinding.rbac.authorization.k8s.io/cluster-autoscaler created
> module.kube-hetzner.null_resource.configure_autoscaler[0] (remote-exec): deployment.apps/cluster-autoscaler created
> module.kube-hetzner.null_resource.configure_autoscaler[0]: Creation complete after 3s [id=2555773439821223472]
> ╷
> │ Error: attach server to network: provided IP is not available (ip_not_available)
> │ 
> │   with module.kube-hetzner.module.agents["1-0-agent-small"].hcloud_server_network.server,
> │   on .terraform/modules/kube-hetzner/modules/host/main.tf line 128, in resource "hcloud_server_network" "server":
> │  128: resource "hcloud_server_network" "server" {
> │ 
> ╵
> ╷
> │ Error: attach server to network: provided IP is not available (ip_not_available)
> │ 
> │   with module.kube-hetzner.module.agents["2-0-agent-small"].hcloud_server_network.server,
> │   on .terraform/modules/kube-hetzner/modules/host/main.tf line 128, in resource "hcloud_server_network" "server":
> │  128: resource "hcloud_server_network" "server" {
> │ 
> ╵

My kube.tf file :
```
locals {
  hcloud_token = "----------------------------------------------------------------"
}

module "kube-hetzner" {
  providers = {
    hcloud = hcloud
  }
  hcloud_token = var.hcloud_token != "" ? var.hcloud_token : local.hcloud_token

  source = "kube-hetzner/kube-hetzner/hcloud"

  ssh_port = 4444

  ssh_public_key = file("~/.ssh/----------------------")
  ssh_private_key = null

  network_region = "eu-central"

  control_plane_nodepools = [
    {
      name        = "control-plane-fsn1",
      server_type = "cpx11",
      location    = "fsn1",
      labels      = [],
      taints      = [],
      count       = 1
      backups = true
    },
    {
      name        = "control-plane-nbg1",
      server_type = "cpx11",
      location    = "nbg1",
      labels      = [],
      taints      = [],
      count       = 1
      backups = true
    },
    {
      name        = "control-plane-hel1",
      server_type = "cpx11",
      location    = "hel1",
      labels      = [],
      taints      = [],
      count       = 1,
      backups = true
    }
  ]

  agent_nodepools = [
    {
      name        = "agent-small",
      server_type = "cpx11",
      location    = "nbg1",
      labels      = [],
      taints      = [],
      count       = 1,
      longhorn_volume_size = 0
    },
    {
      name        = "agent-small",
      server_type = "cpx11",
      location    = "hel1",
      labels      = [],
      taints      = [],
      count       = 1,
      longhorn_volume_size = 0
    },
    {
      name        = "agent-small",
      server_type = "cpx11",
      location    = "fsn1",
      labels      = [],
      taints      = [],
      count       = 1,
      longhorn_volume_size = 0
    }
  ]
  
  enable_wireguard = true

  load_balancer_type     = "lb11"
  load_balancer_location = "fsn1"

  autoscaler_nodepools = [
    {
      name        = "agent-autoscaled"
      server_type = "cpx11"
      location    = "fsn1"
      min_nodes   = 0
      max_nodes   = 5
    }
  ]

  enable_longhorn = true

  traefik_additional_options = [
    "--global.sendanonymoususage=false"
  ]

  cluster_name = "cluster-1"

  cni_plugin = "cilium"

  use_control_plane_lb = true

  cilium_values = <<EOT
ipam:
  mode: kubernetes
devices: "eth1"
k8s:
  requireIPv4PodCIDR: true
kubeProxyReplacement: strict
l7Proxy: false
encryption:
  enabled: true
  type: wireguard
  EOT
}

provider "hcloud" {
  token = var.hcloud_token != "" ? var.hcloud_token : local.hcloud_token
}

terraform {
  required_version = ">= 1.3.3"
  required_providers {
    hcloud = {
      source  = "hetznercloud/hcloud"
      version = ">= 1.38.2"
    }
  }
}

output "kubeconfig" {
  value     = module.kube-hetzner.kubeconfig
  sensitive = true
}

variable "hcloud_token" {
  sensitive = true
  default   = ""
}
```

**Comment by mysticaltech** [comment]
@K-MeLeOn It was probably an issue with Hetzner at the time, please try again.

**Reply by K-MeLeOn** [reply]
Mmh, I've tried several times and it doesn't work.
I think it's a real bug, so let me explain.

The problem only affects the control-plane and the node-agent, not the egress node.

When you choose the same AZ for the control-planes or the node-agents, there's no network problem, but when you choose another AZ, for example two cp on ngb1 and one cp on fsn1, fsn1 doesn't connect to the network.
When I choose three on the same zone, everything's fine. I can have three control-planes on nbg1 and three node-agents on fsn1 without any problem.

I don't understand why, because the egress node, added at the end of the node list, can use another AZ and successfully connect to the private network.

**Reply by mysticaltech** [reply]
@K-MeLeOn I just understood. You use the same names for agent nodepools, each must have a unique name!

**Reply by K-MeLeOn** [reply]
Ooh, as I saw that creation adds extra chars to avoid duplicate names, I didn't think that could be the problem!
I'll try later, thanks for your reply!
